internal/netsync/manager.go

```diff
@@ -474,98 +474,96 @@ func (m *SyncManager) fetchNextHeaders(peer *Peer) {
+		if peer.LastBlock() < bestHeight {
 			peer.syncCandidate = false
 			continue
 		}
 
 		// The best sync candidate is the most updated peer.
-		if bestPeer == nil {
+		if bestPeer == nil || bestPeer.LastBlock() < peer.LastBlock() {
 			bestPeer = peer
 		}
-		if bestPeer.LastBlock() < peer.LastBlock() {
-			bestPeer = peer
-		}
-	}
-
-	// Update the state of whether or not the manager believes the chain is
-	// fully synced to whatever the chain believes when there is no candidate
-	// for a sync peer.
-	//
-	// Also, return now when there isn't a sync peer candidate as there is
-	// nothing more to do without one.
-	if bestPeer == nil {
-		m.isCurrentMtx.Lock()
-		m.isCurrent = chain.IsCurrent()
-		m.isCurrentMtx.Unlock()
-		log.Warnf("No sync peer candidates available")
-		return
 	}
+	m.syncPeer = bestPeer
+}
 
-	// Start syncing from the best peer.
-
-	// Clear the requestedBlocks if the sync peer changes, otherwise
-	// we may ignore blocks we need that the last sync peer failed
-	// to send.
-	m.requestedBlocks = make(map[chainhash.Hash]struct{})
-
-	syncHeight := bestPeer.LastBlock()
-
-	headersSynced := m.hdrSyncState.headersSynced
-	if !headersSynced {
-		log.Infof("Syncing headers to block height %d from peer %v", syncHeight,
-			bestPeer)
-	}
+// startInitialHeaderSync starts the initial header sync process which consists
+// of downloading all headers that are not already known from the sync peer. As
+// the name implies, this assumes the initial header sync process is not already
+// done.
+func (m *SyncManager) startInitialHeaderSync(bestHeight int64) {
+	hdrSyncPeer := m.syncPeer
+	syncHeight := hdrSyncPeer.LastBlock()
+	log.Infof("Syncing headers to block height %d from peer %v", syncHeight,
+		hdrSyncPeer)
 
 	// The chain is not synced whenever the current best height is not within a
 	// couple of blocks of the height to sync to.
-	if best.Height+2 < syncHeight {
+	if bestHeight+2 < syncHeight {
 		m.isCurrentMtx.Lock()
 		m.isCurrent = false
 		m.isCurrentMtx.Unlock()
 	}
 
 	// Request headers starting from the parent of the best known header for the
 	// local chain from the sync peer.
-	m.fetchNextHeaders(bestPeer)
+	m.fetchNextHeaders(hdrSyncPeer)
 
-	// Track the sync peer and update the sync height when it is higher than the
-	// currently best known value.
-	m.syncPeer = bestPeer
+	// Update the sync height when it is higher than the currently best known
+	// value.
 	m.syncHeightMtx.Lock()
 	if syncHeight > m.syncHeight {
 		m.syncHeight = syncHeight
 	}
 	m.syncHeightMtx.Unlock()
 
-	// Start the header sync progress stall timeout when the initial headers
-	// sync is not already done.
+	// Start the header sync progress stall timeout.
+	m.hdrSyncState.resetStallTimeout()
+}
+
+// startSync will choose the best peer among the available candidate peers to
+// download/sync the blockchain from.  It also examines the candidates for any
+// which are no longer candidates and removes them as needed.
```

Comment:
Same comment here as above. Peers are not being disconnected, so the "removed" wording is ambiguous/confusing.

Suggestion:

```
// startSync selects the peer with the most blocks and begins to download the
// blockchain headers from them. It also examines the sync candidates and marks
// those whose chain heights have been surpassed as ineligible to sync from.
```

Reply:
Hmm.  I agree with the potential ambiguity around "removed", but the rest of the comment is questionable since it really does more than just the blockchain headers depending on the state of things.

Reply:
I updated it clarify a bit.

Reply:
I updated it clarify a bit.  Does the new text make things more clear?

Reply:
Yep, thanks for clarifying!